Title: IBM Inbound event is delivered twice by adapter - United States

Text:
wasapv602rnote; wasblv602rnote; wajdbcv602rnote; wajdev602rnote; waffv602rnote; waftpv602rnote; waemlv602rnote; waorclv602rnote; wasapv61rnote; wasblv61rnote; wajdbcv61rnote; wajdev61rnote; waffv61rnote; waftpv61rnote; waemlv61rnote; waorclv61rnote; watoolkitv61rnote; wamigrt61rnote; wapsftv61rnote TECHNOTE (TROUBLESHOOTING)

PROBLEM(ABSTRACT)
 When synchronizing data from Siebel to SAP, the Siebel adapter delivers Siebel events to an endpoint in WebSphere Process Server. A relationship manages the cross references of SAP and Siebel event keys. The Siebel inbound event is sent to the Adapter for SAP and finally the data is synced to SAP. Under certain configurations, if the Siebel server is rebooted while Siebel inbound events are being processed, the same event may be delivered twice. 

CAUSE
Using the scenario just described as a reference, configure the Siebel to SAP data sync end-to-end integration as follows: 

 * Create an SOA application that can synchronize data from Siebel to SAP using WebSphere Adapters and WebSphere Process Server. (Information on how to do this can be found in the WebSphere Adapters information center and IBM Developer Works) 
 * In the Siebel adapter export, set Assured Once Delivery to "true" and the interaction style to "Sync." 
 * Create a large number of Siebel inbound events. 
 * While the Siebel adapter is processing inbound events, reboot the Siebel server.


When the Siebel server is restarted, the Siebel adapter will re-deliver the event that was being processed by the relationship when the Siebel server was shut down. This causes the relationship to generate an exception because it already has an entry with the same Siebel key in the relationship database. This is actually not a bug. Because the interaction style was set to Sync, the inbound process ends when the event is delivered to SAP and returned to the adapter. The adapter will then update the event table in Siebel. However, it cannot get to it as the Siebel server was shut down. When the Siebel server restarts, the adapter will re-deliver the same event because the event table shows that this event was not successfully delivered. 
RESOLVING THE PROBLEM
To resolve this problem, set the Siebel adapter export the interaction style property to Async. Note, however, this may cause a problem if you performing an immediate update to a object that was just created in Siebel. The Async mode will cause the create and update events to be processed in parallel. If an update event reaches the relationship before a create event is completed, the relationship will generate a key not found exception. 

To resolve this, add event sequencing QoS to the relationship. Doing this guarantees that the create event is completed before the update event is processed. 

In summary, to solve the problem, ensure that you do the following: 

 * Set the interaction style for inbound to Async. 
 * Set Assure Once Delivery to "true." 
 * Add event sequencing QoS to the relationship.